nft-example/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+            .balances
+            .get(&msg::source())
+            .unwrap_or(&U256::zero());
+        self.tokens
+            .balances
+            .insert(msg::source(), balance.saturating_add(U256::one()));
+
+        let transfer_token = Transfer {
+            from: H256::zero(),
+            to: H256::from_slice(msg::source().as_ref()),
+            token_id: self.token_id,
+        };
+
+        self.token_id = self.token_id.saturating_add(U256::one());
+
+        msg::reply(
+            Event::Transfer(transfer_token),
+            exec::gas_available() - GAS_RESERVE,
+            0,
+        );
+    }
+
+    fn burn(&mut self, token_id: U256) {
+        if !self.tokens.exists(token_id) {
+            panic!("NonFungibleToken: Token does not exist");
+        }
+        if !self.tokens.is_token_owner(token_id, &msg::source()) {
+            panic!("NonFungibleToken: account is not owner");
+        }
+        self.tokens.token_approvals.remove(&token_id);
+        self.tokens.owner_by_id.remove(&token_id);
+        let balance = *self
+            .tokens
+            .balances
+            .get(&msg::source())
+            .unwrap_or(&U256::zero());
+        self.tokens
+            .balances
+            .insert(msg::source(), balance.saturating_sub(U256::one()));
+
+        let transfer_token = Transfer {
+            from: H256::from_slice(msg::source().as_ref()),
+            to: H256::zero(),
+            token_id,
+        };
+        msg::reply(
+            Event::Transfer(transfer_token),
+            exec::gas_available() - GAS_RESERVE,
+            0,
+        );
+    }
+}
+
+gstd::metadata! {
+    title: "NftExample",
+        init:
+            input: InitConfig,
+        handle:
+            input: Action,
+            output: Event,
+        state:
+            input: State,
+            output: StateReply,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let action: Action = msg::load().expect("Could not load Action");
+    match action {
+        Action::Mint => {
+            CONTRACT.mint();
+        }
+        Action::Burn(input) => {
+            CONTRACT.burn(input);
+        }
+        Action::Transfer(input) => {
+            CONTRACT.tokens.transfer(
+                &msg::source(),
+                &ActorId::new(input.to.to_fixed_bytes()),
+                input.token_id,
+            );
+        }
+        Action::Approve(input) => {
+            CONTRACT.tokens.approve(
+                &msg::source(),
+                &ActorId::new(input.to.to_fixed_bytes()),
+                input.token_id,
+            );
+        }
+        Action::ApproveForAll(input) => {
+            CONTRACT.tokens.approve_for_all(
+                &msg::source(),
+                &ActorId::new(input.to.to_fixed_bytes()),
+                input.approve,
+            );
+        }
+        Action::OwnerOf(input) => {
+            let owner = CONTRACT.tokens.owner_by_id.get(&input).unwrap_or(&ZERO_ID);
```

Comment:
When the interface is updated, please replace this with `CONTRACT.tokens.owner_of` or similar.

Reply:
Updated interface and example, please check